app/queues/notifications/certificate-expiration-monitor.server.ts

```diff
@@ -0,0 +1,59 @@
+import { Queue } from 'bullmq';
+import { redis } from '~/lib/redis.server';
+import logger from '~/lib/logger.server';
+import type { DomainStatusData } from './record-cert-expiration-data.interface';
+import secrets from '~/lib/secrets.server';
+import { prisma } from '~/db.server';
+import {
+  startCertificateExpirationMonitorWorker,
+  certExpirationQueueName,
+} from './certificate-expiration-monitor-worker.server';
+
+// fetch records with by status
+const getCertificatesByExpiration = async () => {
+  let monthFromToday = new Date();
+  monthFromToday.setMonth(monthFromToday.getMonth() + 1);
+  return await prisma.certificate.findMany({
+    where: {
+      AND: [
+        {
+          validTo: {
+            lte: monthFromToday,
+          },
+          isNotified: false,
+        },
+      ],
+    },
+    select: {
+      user: true,
+      id: true,
+    },
+  });
+};
+
+export const certificateExpirationCheckQueue = new Queue<DomainStatusData[]>(
+  certExpirationQueueName,
+  {
+    connection: redis,
+    defaultJobOptions: {
+      attempts: 3,
+      backoff: {
+        type: 'exponential',
+        delay: 60 * 1000 /* 1 minute */,
+      },
+    },
+  }
+);
+
+certificateExpirationCheckQueue.on('error', (err) => {
+  logger.warn('Error running check for domains about to expire', err);
+});
+
+export const addCertificateExpirationCheckJobs = async () => {
+  let jobName = `${certExpirationQueueName}:${new Date().toLocaleString('en-CA')}`;
+  return await certificateExpirationCheckQueue.add(jobName, await getCertificatesByExpiration(), {
+    repeat: { pattern: secrets.NOTIFICATIONS_REPEAT_PATTERN },
```

Comment:
I don't think that this needs to be a secret, I think it;s quite enough to either hardcode it, or use an env

Reply:
I think it would be better to set it as env so it's easier to update the value if needed.